File: DictionaryProject/Commands/CommandCollection.vb

Public Class CommandCollection
    Inherits Collections.ObjectModel.Collection(Of CommandBase)

    Public Sub Execute(ByVal context As Context)
        For Each cmd As CommandBase In Me
            cmd.Execute(context)
        Next
    End Sub

    Public Sub WriteCommands(ByVal writer As IO.TextWriter)
        For Each c As CommandBase In Me
            c.WriteCommand(writer)
        Next
    End Sub
End Class

File: monowordbuilder/wordbuilderbase/Commands/CommandCollection.cs
using System;
using System.IO;

namespace Whee.WordBuilder.Project.Commands
{
	public class CommandCollection : System.Collections.ObjectModel.Collection<CommandBase>
	{
		public void Execute(Context context)
		{
			foreach (CommandBase cmd in this) {
				cmd.Execute(context);
			}
		}
	    
		public void WriteCommands(TextWriter writer)
	    {
	        foreach (CommandBase c in this) {
	            c.WriteCommand(writer);
	        }
	    }
	}
}